addons/web/static/src/views/fields/properties/property_value.js

```diff
@@ -0,0 +1,290 @@
+            if (typeof value === 'string') {
+                // convert the datetime from the UTC format to the current timezone
+                const datetimeValue = DateTime.fromFormat(
+                    value + ' +00:00',
+                    DATETIME_FORMAT + ' Z',
+                );
+                return datetimeValue.invalid ? false : datetimeValue;
+            }
+            return (value instanceof DateTime) ? value : false;
+        } else if (this.props.type === 'date') {
+            if (typeof value === 'string') {
+                const datetimeValue = DateTime.fromFormat(
+                    value + ' +00:00',
+                    DATE_FORMAT + ' Z',
+                );
+                return datetimeValue.invalid ? false : datetimeValue;
+            }
+            return (value instanceof DateTime) ? value : false;
+        } else if (this.props.type === 'boolean') {
+            return !!value;
+        } else if (this.props.type === 'selection') {
+            const options = this.props.selection || [];
+            const option = options.find(option => option[0] === value);
+            return option && option.length === 2 && option[0] ? option[0] : '';
+        } else if (this.props.type === 'many2one') {
+            if (!value || value.length !== 2 || !value[0]) {
+                return false;
+            }
+            return value;
+        } else if (this.props.type === 'many2many') {
+            if (!value || !value.length) {
+                return [];
+            }
+
+            // Convert to TagList format
+            return value.map((many2manyValue) => {
+                return {
+                    'id': many2manyValue[0],
+                    'text': many2manyValue[1],
+                    'onClick': () => this._openRecord(this.props.model, many2manyValue[0]),
+                    'onDelete': !this.props.readonly && (() => this.onMany2manyDelete(many2manyValue[0])),
+                    'colorIndex': 0,
+                };
+            });
+        } else if (this.props.type === 'tags') {
+            return value || [];
+        }
+
+        return value;
+    }
+
+    /**
+     * Return the model domain (related to many2one and many2many properties).
+     */
+    get propertyDomain() {
+        if (!this.props.domain || !this.props.domain.length) {
+            return [];
+        }
+        return new Domain(this.props.domain).toList();
+    }
+
+    /**
+     * Formatted value displayed in readonly mode.
+     */
+    get displayValue() {
+        const value = this.propertyValue;
+
+        if (this.props.type === 'many2one' && value && value.length === 2) {
+            return `${value[1]} (#${value[0]})`;
+        } else if (!value) {
+            return false;
+        } else if (this.props.type === 'datetime' && value) {
+            return value.toFormat('LL/dd/yyyy HH:mm:ss');
+        } else if (this.props.type === 'date' && value) {
+            return value.toFormat('LL/dd/yyyy');
```

Comment:
Hmm not sure about those, should use user locale defined in Odoo settings right?

Again, maybe we have tools to format values?